src/pint/residuals.py

```diff
@@ -107,7 +108,7 @@ def __init__(
         self.toas = toas
         self.model = model
         self.residual_type = residual_type
-        self.subtract_mean = subtract_mean
+        self.subtract_mean = subtract_mean and "PhaseOffset" not in model.components
```

Comment:
Maybe a `debug` message when the presence of a PhaseOffset overrides `subtract_mean`?

Reply:
added debug message